### ISSUE TITLE
Problem with ContextMenu in ListView, cannot retrieve position from AdapterContextMenuInfo 

### ISSUE THREAD

**Issue by ForestTree** [Prototik/HoloEverywhere#178]
Hello,
I have problem when trying read "id" or "position" fields from AdapterContextMenuInfo. This fields are not initialized for ContextMenu in ListView. So I cannot get position in the adapter where was ContextMenu invoked. Here is short example:

<code>
public class MainActivity extends SActivity {

```
private ListView mList;     

@Override
public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    setContentView(R.layout.activity_main);

    mList = (ListView) findViewById(R.id.listView1); 
    String[] values = new String[] { "Alpha", "Bravo", "Charlie", "Delta" };

    ArrayAdapter<String> adapter = new ArrayAdapter<String>(this,
            android.R.layout.simple_list_item_1, android.R.id.text1, values);

    mList.setAdapter(adapter); 
    mList.setOnCreateContextMenuListener(this);        
}

@Override
public boolean onCreateOptionsMenu(Menu menu) {
    getSupportMenuInflater().inflate(R.menu.activity_main, menu);
    return true;
}

@Override
public void onCreateContextMenu(ContextMenu menu, View view, ContextMenuInfo menuInfo) {
    getSupportMenuInflater().inflate(R.menu.context_menu, menu);
} 

@Override
public boolean onContextItemSelected(MenuItem item) {   
    final AdapterContextMenuInfo info = (AdapterContextMenuInfo) item.getMenuInfo();
    // !!!!! here is problem:
    long id = info.id;
    int pos = info.position;
    return super.onContextItemSelected(item);
}
```

}
</code>

Here is exception:

<exception>
11-02 11:39:02.359: E/AndroidRuntime(4173): FATAL EXCEPTION: main
11-02 11:39:02.359: E/AndroidRuntime(4173): java.lang.NullPointerException
11-02 11:39:02.359: E/AndroidRuntime(4173):     at com.example.helistexample.MainActivity.onContextItemSelected(MainActivity.java:51)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at com.actionbarsherlock.internal.view.menu.ContextMenuDecorView$InternalWrapper.onMenuItemSelected(ContextMenuDecorView.java:47)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at com.actionbarsherlock.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:738)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at com.actionbarsherlock.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:148)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at com.actionbarsherlock.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:879)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at com.actionbarsherlock.internal.view.menu.MenuDialogHelper.onClick(MenuDialogHelper.java:34)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at org.holoeverywhere.internal.AlertController$AlertParams$3.onItemClick(AlertController.java:221)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at android.widget.AdapterView.performItemClick(AdapterView.java:298)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at android.widget.AbsListView.performItemClick(AbsListView.java:1086)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at org.holoeverywhere.widget.ListView.performItemClick(ListView.java:170)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at android.widget.AbsListView$PerformClick.run(AbsListView.java:2859)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at android.widget.AbsListView$1.run(AbsListView.java:3533)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at android.os.Handler.handleCallback(Handler.java:615)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at android.os.Handler.dispatchMessage(Handler.java:92)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at android.os.Looper.loop(Looper.java:137)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at android.app.ActivityThread.main(ActivityThread.java:4745)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at java.lang.reflect.Method.invokeNative(Native Method)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at java.lang.reflect.Method.invoke(Method.java:511)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
11-02 11:39:02.359: E/AndroidRuntime(4173):     at dalvik.system.NativeStart.main(Native Method)
</exception>

**Comment by Prototik** [comment]
Try dev version.

**Comment by ForestTree** [comment]
I downloaded actual dev branch and still the same problem.

**Comment by Prototik** [comment]
Come on?
Test application after launch:
![Screen 1](http://i.imgur.com/uEk69.png)
Long press on items:
![Screen 2](http://i.imgur.com/1ocJZ.png)
![Screen 3](http://i.imgur.com/MdRpT.png)

Source:

``` java
package k.listviewtest;

import org.holoeverywhere.sherlock.SListActivity;

import android.os.Bundle;
import android.view.ContextMenu.ContextMenuInfo;
import android.view.View;
import android.widget.AdapterView.AdapterContextMenuInfo;
import android.widget.ArrayAdapter;

import com.actionbarsherlock.view.ContextMenu;

public class ListViewTestActivity extends SListActivity {
    private final String[] s = {
            "Item 1", "Item 2", "Item 3", "Item 4", "Item 5"
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        registerForContextMenu(getListView());
        setListAdapter(new ArrayAdapter<String>(this, R.layout.simple_spinner_item,
                android.R.id.text1, s));
    }

    @Override
    public void onCreateContextMenu(ContextMenu menu, View view, ContextMenuInfo menuInfo) {
        getSupportActionBar().setSubtitle(
                "Position: " + ((AdapterContextMenuInfo) menuInfo).position);
    }
}
```

**Comment by ForestTree** [comment]
Hello, 
thanks for prompt fix. Redesigned to SListActivity, no problem. AdapterContextMenuInfo obtained on onCreateContextMenu contains valid "id" and "position".
And last question, when I try:

<code>
public boolean onContextItemSelected(MenuItem item) {
    AdapterContextMenuInfo info = (AdapterContextMenuInfo) item.getMenuInfo();
}
</code>

I receive null AdapterContextMenuInfo. Is this correct or still some bug?
Thanks.

**Comment by Prototik** [comment]
Bug, already fixed. Update HE.

**Comment by ForestTree** [comment]
Many thanks Sergey, work like a charm.

**Comment by 4nd3250n** [comment]
What version has the bug fixed? I use 1.3.3 and have the same problem.

**Comment by Prototik** [comment]
@4nd3250n dev version
